text/0000-transaction-bundle/0000-transaction-bundle.md

```diff
@@ -0,0 +1,569 @@
++ Feature name: (fill me in with a unique ident, `my_awesome_feature`)
++ Start date: 2019-09-06
++ RFC PR: [iotaledger/bee-rfcs#3](https://github.com/iotaledger/bee-rfcs/pull/3),
+[iotaledger/bee-rfcs#18](https://github.com/iotaledger/bee-rfcs/pull/18),
+[iotaledger/bee-rfcs#20](https://github.com/iotaledger/bee-rfcs/pull/20)
++ Bee issue: [iotaledger/bee#0000](https://github.com/iotaledger/bee/issues/0000)
+
+# Summary
+
+## Transaction
+
+In the context of IOTA, a `Bundle` is a message that is shared over the network, and a `Transaction` is the fundamental
+unit that a `Bundle` is constructed from. To use TCP as a well known analogy, a `Bundle` corresponds to a message, and a `Transaction` corresponds to a packet.
+
+This RFC proposes a `Transaction` type to represent the transaction format used by the IOTA Reference Implementation as
+of version [`iri v1.8.1`], commit `e1776fbad5d90df86a26402f9025e4b0b2ef7d3e`. The `Transaction` type is implemented as a
+struct and is intended to be a public interface. It thus comes with a number of opaque types for its fields that we
+leave to be fleshed out at a later time.
+
+We also propose a `TransactionDraft` type that follows the builder pattern. It is intended to be an implementation
+detail and only be used by a `Bundle` builder type and not exposed publicly.
+
+## Bundle
+
+The smallest communication unit in the IOTA protocol is the transaction. Everything, including payment settlements
+and/or plain data, is propagated through the IOTA network in transactions.
+
+A transaction is `8019` trits and the payload (i.e. `sig_or_msg`) is `6561` trits. This payload holds a signature in
+case of a payment settlement and plain data otherwise. Since it has a limited size, a user often needs more than one
+transaction to fulfil their operation, for example signatures with security level `2` or `3` don't fit in a single
+transaction and user-provided data may exceed the allowance so they need to be fragmented across multiple transactions.
+Moreover, a value transaction doesn't make sense on its own because it would change the total amount of the ledger so
+it has to be paired with other complementary transactions that together will balance the total value to zero.
+
+For these reasons, transactions have to be processed as a whole, in groups called bundles. A bundle is an atomic
+operation in the sense that either all or none of its transactions are accepted by the network. Even single
+transactions are propagated through the network within a bundle making it the only confirmable communication unit of
```

Comment:
Isn't the term `confirmation/confirmed` bit meaningless in an open network? Also Paul Handy, as well as CfB - if I remember correctly - don't like it.

Reply:
I don't think it's meaningless in the sense it's being used. The only unit to which consensus can apply is the bundle.